### ISSUE TITLE
Redirects for other sites than www.bethel.edu

### ISSUE THREAD

**Issue by c-schwarze** [betheluniversity/tinker#261]
We want to change redirects so they can be added to other domains than www.bethel.edu. (mybethel.edu, etc.).

It might be worth upkeeping a select field so that when redirects are added, they don't have to be manually typed with the potential of spelling errors. I think we can still use the same redirects.txt file. 

@bak45247 This can be your next project, as you already have been working on redirects.

**Comment by e-jameson** [comment]
We would need one file for each domain. 

We could also look into loading the redirects from a DB instead of txt file on RHEL7 servers. I believe the Apache version on RHEL7 supports that feature.

**Comment by c-schwarze** [comment]
Bummer. I assume we just add more lines of these? I'm a little confused how this will be able to affect other domains on other vhosts (like classifieds).

```
RewriteMap hostmap txt:/opt/tinker/tinker/redirects.txt
RewriteCond ${hostmap:%{REQUEST_FILENAME}} !^$
RewriteRule .* ${hostmap:%{REQUEST_FILENAME}} [R=301,L]
```

I would propose we wait to load redirects from a DB, until Tinker is on a RHEL7 box.

**Comment by e-jameson** [comment]
If you look at the file, the format for each line is for a `RewriteMap` file is

`/id-cards https://confluence.bethel.edu/x/AwAt`

The left side is the path that should be redirected, the right side is where it should go. This is why the format of the Tinker Redirects module is how it is. The domain is assumed to be whatever domain Apache is currently dealing with. So using the same file means that all www.bethel.edu redirects would also work on my.bethel.edu, etc.

Loading from a DB wouldn't be all that much different. We would need a DB per domain instead of a txt file per domain. So I'm not sure we'd gain that much by waiting.

**Comment by c-schwarze** [comment]
Sure. But how do we link from h13's apache to h24's mybethel-redirects.txt file? Also it looks like /opt/tinker/tinker/redirects.txt is empty? Does apache consume the file?

Again, I think I am just misunderstanding you. I thought you were saying that apache on RHEL7 supports loading from a DB. So my concern was having our redirects needing to work for both a redirects.txt file and a DB (depending on the domain).

**Comment by e-jameson** [comment]
The path for the redirects file is `/var/www/cms.pub/redirects/redirects.txt`. It's in cms.pub so it can stay in sync between h24 and h20. 

You are right, we'd have to use sftp or something to sync a txt or db-type file from h24 to the other servers that need a redirects file. Check out all the options:

https://httpd.apache.org/docs/trunk/rewrite/rewritemap.html#page-header

It's possible we could use a DB query to store them all in one database. Something like a DBM file we'd have to sync, but still has advantages over a text file.

If I'm still explaining it bad we can draw it out sometime, maybe our next Monday meeting.

**Comment by c-schwarze** [comment]
That makes sense. I am all for getting rid of the .txt file. I think its worth pursuing and it might be a great topic for Monday, just cause.

**Comment by c-schwarze** [comment]
I kind of like the idea of using a single db, but use a mysql query.

**Comment by c-schwarze** [comment]
@j-jasik you can get started on this. You should be able to do a good amount of the work before we decide how we want to output the data.

Feel free to read eric and i's conversation above.

**Comment by c-schwarze** [comment]
#294 is the PR for this

**Comment by c-schwarze** [comment]
We are just waiting on Sysadmin to get us a good way to move the files regularly. IT FP case # 188656

**Comment by c-schwarze** [comment]
We are able to SFTP between our servers easily. We should close case # 188656 when we finish this.

@j-jasik this is still very low priority, but just an FYI its still here.
